include/ArgumentParser.h

```diff
@@ -0,0 +1,30 @@
+#ifndef ARGUMENT_PARSER_H
+#define ARGUMENT_PARSER_H
+
+#include "ErrorCodes.h"
+#include <stdint.h>
+#include <expected>
+namespace Chat
+{
+    enum class ProgramType
+    {
+        CLIENT,
+        SERVER
+    };
+
+    typedef struct
+    {
+        uint32_t address;
+        uint16_t port;
+    } ClientInfo;
```

Comment:
If you want to be technical, this struct should be called `Address` and the inner field should be called `ip`